Add manual "Add Pin" dialog and File→Add Pin menu; update player pin …

amnindersingh12 opened the pull request:
Add manual "Add Pin" feature (Closes #40)

- Add modal dialog to add pins by hand: Forms/FrmAddPin.cs
- Add "Add Pin..." menu item in File menu and wire click handler:
  - Forms/FrmMain.Designer.cs (add addPinToolStripMenuItem)
  - Forms/FrmMain.cs (AddPinToolStripMenuItem_Click to show dialog, update playerObject lists, update counts, avoid duplicates, enable Save)
- Include new form in project file: SoG_Savegame_Editor.csproj
- Fix enum name collision by qualifying System.Enum usage in FrmAddPin.cs

Behavior:
- User can open File → Add Pin..., select a SogPin and target list (PinsSeen, PinsOnShelf, PinsEquipped, PinsLatest).
- Selected pin is added to the chosen in-memory player list (lazy-init if null), duplicates prevented, counts updated, Save enabled.
 
Fixes: https://github.com/tolik518/SoG-Savegame-Editor/issues/40

amnindersingh12 committed:
Add manual "Add Pin" dialog and File→Add Pin menu; update player pin lists and project

amnindersingh12 committed:
refactored AddPinToPlayer

amnindersingh12 committed:
Removed FrmAddPin.cs file
Removed "Add Pin..." menu item from File menu
Removed helper methods: AddPinToolStripMenuItem_Click, ShowAddPinDialog, AddPinToPlayer, PostAddPinUpdate
Updated .csproj to exclude the deleted file

amnindersingh12 commented:
@tolik518 Made some chages please have a look at it !!

tolik518 commented:
@amnindersingh12  Looks good to me on the first glance :)
The only thing that's now missing is the following:

If the user marks an item as equipped, it should be automatically be added as seen (you cant equip something you've never seen). Otherwise it looks good and I'll merge after our adjustment!

amnindersingh12 committed:
- Add code to save equipment data in GetDataFromFields()
- Add a helper method to mark an item as seen
- Add event handlers to equipment combo boxes
